### ISSUE TITLE
CPU Performance

### ISSUE THREAD

**Issue by Tomazz** [flask-dashboard/Flask-MonitoringDashboard#200]
So I installed the dashboard and it started chewing through pretty much all the CPU credits I had. The CPU usage was constantly at 100%. 

![screenshot 41](https://user-images.githubusercontent.com/3969552/47100928-997a6a80-d230-11e8-91e8-fac3279b7c02.png)

**Comment by bogdanp05** [comment]
Hi @Tomazz ,

Thanks for reporting the issue. Could you please check what monitoring level your endpoints were set to inside the dashboard? You can see this in the **Rules** tab, in the column **Monitoring-level**. 

Thanks,
Bogdan

**Comment by Jeremy-F** [comment]
Same "bug" there with Monitoring-level at 3. 

External config used : 

```[dashboard]
APP_VERSION=1.0
CUSTOM_LINK=dashboard
MONITOR_LEVEL=3
OUTLIER_DETECTION_CONSTANT=2.5
SAMPLING_PERIOD=20
ENABLE_LOGGING=True

[authentication]
USERNAME=XXX
PASSWORD=XXX
GUEST_USERNAME=XXX
GUEST_PASSWORD=['XXX', 'XXX!']
SECURITY_TOKEN=ABCD
```

**Comment by bogdanp05** [comment]
Hi @Jeremy-F,

Thanks for reporting the issue. Monitoring levels 2 and 3 include the profiler, which is an extra thread that every `SAMPLING_PERIOD` milliseconds takes a snapshot of the current stack trace and then stores this information to the database. Therefore, a higher CPU usage is to be expected.

We are currently working on reducing the overhead of the profiler. In the meantime, I would use monitoring levels 2 and 3 only for the endpoints that you really need profiled, and use level 1 for the others.

Bogdan

**Comment by bogdanp05** [comment]
Fixed in #259
